File: frontend/src/utils/newsSentiment.js
// 백엔드 NewsSentiment enum과 화면에 표시할 한국어 용어를 한곳에서 관리한다.
export const NEWS_SENTIMENT_LABELS = {
  POSITIVE: "호재",
  NEUTRAL: "보통",
  NEGATIVE: "악재"
};

// 감성값이 없는 이전 캐시는 보통으로 계산해 기사 수와 요약 집계의 합계를 일치시킨다.
export function normalizeNewsSentiment(sentiment) {
  return NEWS_SENTIMENT_LABELS[sentiment] ? sentiment : "NEUTRAL";
}

export function countNewsSentiments(items) {
  return items.reduce((counts, item) => {
    counts[normalizeNewsSentiment(item.sentiment)] += 1;
    return counts;
  }, { POSITIVE: 0, NEUTRAL: 0, NEGATIVE: 0 });
}

File: src/main/java/com/finmate/domain/news/NewsSentiment.java
package com.finmate.domain.news;

/**
 * 종목 뉴스와 시장 리포트에 표시할 세 단계 감성 결과다.
 * KR-FinBert-SC의 positive/neutral/negative 라벨을 화면 용어인 호재/보통/악재로 대응한다.
 */
public enum NewsSentiment {
    POSITIVE,
    NEUTRAL,
    NEGATIVE
}
